classes/PodsData.php

```diff
@@ -1998,12 +1998,11 @@ public function fetch( $row = null, $explicit_set = true ) {
 			$mode = 'id';
 			$id   = pods_absint( $row );
 
-			if ( ! is_numeric( $row ) || 0 === strpos( $row, '0' ) || $row != preg_replace( '/[^0-9]/', '', $row ) ) {
+			if ( null !== $row && ( ! is_numeric( $row ) || 0 === strpos( $row, '0' ) || $row != preg_replace( '/[^0-9]/', '', $row ) ) ) {
 				if ( $this->id && is_numeric( $this->id ) ) {
 					$id = $this->id;
 				} else {
-					$mode = 'slug';
-					$id   = $row;
+					$id = $this->id;
```

Comment:
This doesn't make sense. Why the if/else if both results are the same?
I think this should be reverted to the original:
```
$mode = 'slug';
$id = $row;
```

Reply:
It has been 3 years since I wrote this, so I cannot tell you other than the patched fixed an issue then and today.

Reply:
When we'll have to make sure this doesn't break anything else before merging.
Also, the code itself would need to update since this if/else statement doesn't make any sense.

Reply:
Anyone else can take over on this then.